### ISSUE TITLE
V6 signUp not returning userId

### ISSUE THREAD

**Issue by jimmysafe** [aws-amplify/amplify-js#12998]
### Before opening, please confirm:

- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-js/issues?q=is%3Aissue+) and [discussions](https://github.com/aws-amplify/amplify-js/discussions).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#bug-reports).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

### JavaScript Framework

React

### Amplify APIs

Authentication

### Amplify Version

v6

### Amplify Categories

auth

### Backend

None

### Environment information

<details>

```
# Put output below this line
  System:
    OS: macOS 13.4
    CPU: (12) arm64 Apple M2 Pro
    Memory: 50.41 MB / 16.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 18.17.1 - ~/.nvm/versions/node/v18.17.1/bin/node
    Yarn: 1.22.19 - /opt/homebrew/bin/yarn
    npm: 9.6.7 - ~/.nvm/versions/node/v18.17.1/bin/npm
    pnpm: 7.32.5 - /opt/homebrew/bin/pnpm
  Browsers:
    Chrome: 121.0.6167.160
    Safari: 16.5
  npmPackages:
    @dnd-kit/core: ^6.1.0 => 6.1.0 
    @dnd-kit/sortable: ^8.0.0 => 8.0.0 
    @dnd-kit/utilities: ^3.2.2 => 3.2.2 
    @hookform/resolvers: ^3.3.4 => 3.3.4 
    @openapi-codegen/cli: ^2.0.0 => 2.0.0 
    @openapi-codegen/typescript: ^8.0.0 => 8.0.0 
    @radix-ui/react-accordion: ^1.1.2 => 1.1.2 
    @radix-ui/react-alert-dialog: ^1.0.5 => 1.0.5 
    @radix-ui/react-avatar: ^1.0.4 => 1.0.4 
    @radix-ui/react-checkbox: ^1.0.4 => 1.0.4 
    @radix-ui/react-dialog: ^1.0.5 => 1.0.5 
    @radix-ui/react-dropdown-menu: ^2.0.6 => 2.0.6 
    @radix-ui/react-icons: ^1.3.0 => 1.3.0 
    @radix-ui/react-label: ^2.0.2 => 2.0.2 
    @radix-ui/react-popover: ^1.0.7 => 1.0.7 
    @radix-ui/react-select: ^2.0.0 => 2.0.0 
    @radix-ui/react-separator: ^1.0.3 => 1.0.3 
    @radix-ui/react-slot: ^1.0.2 => 1.0.2 
    @radix-ui/react-switch: ^1.0.3 => 1.0.3 
    @radix-ui/react-tabs: ^1.0.4 => 1.0.4 
    @radix-ui/react-toast: ^1.1.5 => 1.1.5 
    @radix-ui/react-toggle: ^1.0.3 => 1.0.3 
    @radix-ui/react-tooltip: ^1.0.7 => 1.0.7 
    @tanstack/react-query: ^5.17.10 => 5.17.10 
    @tanstack/react-query-devtools: ^5.17.10 => 5.17.10 
    @tanstack/react-table: ^8.11.8 => 8.11.8 
    @tiptap/core: ^2.2.1 => 2.2.1 
    @tiptap/extension-heading: ^2.2.1 => 2.2.1 
    @tiptap/pm: ^2.2.1 => 2.2.1 
    @tiptap/react: ^2.2.1 => 2.2.1 
    @tiptap/starter-kit: ^2.2.1 => 2.2.1 
    @types/node: ^20.11.0 => 20.11.0 
    @types/react: ^18.2.43 => 18.2.47 
    @types/react-dom: ^18.2.17 => 18.2.18 
    @types/uuid: ^9.0.8 => 9.0.8 
    @typescript-eslint/eslint-plugin: ^6.14.0 => 6.18.1 
    @typescript-eslint/parser: ^6.14.0 => 6.18.1 
    @vitejs/plugin-react: ^4.2.1 => 4.2.1 
    autoprefixer: ^10.4.16 => 10.4.16 
    aws-amplify: ^6.0.16 => 6.0.16 
    class-variance-authority: ^0.7.0 => 0.7.0 
    clsx: ^2.1.0 => 2.1.0 
    date-fns: ^3.3.1 => 3.3.1 
    eslint: ^8.55.0 => 8.56.0 
    eslint-plugin-react-hooks: ^4.6.0 => 4.6.0 
    eslint-plugin-react-refresh: ^0.4.5 => 0.4.5 
    lucide-react: ^0.309.0 => 0.309.0 
    postcss: ^8.4.33 => 8.4.33 
    react: ^18.2.0 => 18.2.0 
    react-day-picker: ^8.10.0 => 8.10.0 
    react-dom: ^18.2.0 => 18.2.0 
    react-hook-form: ^7.49.3 => 7.49.3 
    react-if: ^4.1.5 => 4.1.5 
    react-router-dom: ^6.21.2 => 6.21.2 
    slugify: ^1.6.6 => 1.6.6 
    tailwind-merge: ^2.2.0 => 2.2.0 
    tailwindcss: ^3.4.1 => 3.4.1 
    tailwindcss-animate: ^1.0.7 => 1.0.7 
    typescript: ^5.2.2 => 5.3.3 
    uuid: ^9.0.1 => 9.0.1 
    vite: ^5.0.8 => 5.0.11 
    zod: ^3.22.4 => 3.22.4 
  npmGlobalPackages:
    @aws-amplify/cli: 12.7.1
    corepack: 0.18.0
    npm: 9.6.7

```

</details>

### Describe the bug

When using `signUp` function from `aws-amplify/auth` and the signup is successful, the response does not include `userId` but it only includes the following:

```
isSignUpComplete: true
nextStep: { 
     signUpStep: 'DONE' 
}
```

### Expected behavior

I do not understand why the `userId` is not present in the response. The documentation states it should be there.

### Reproduction steps

install aws-amplify v6
use the signUp function
see that userId is not present in the successful signup response.

### Code Snippet

```javascript
// Put your code below this line.

  static async signUp(data: SignupFormData) {
    const res = await signUp({
      username: data.email,
      password: data.password,
      options: {
        userAttributes: {
          email: data.email.toLowerCase(),
          given_name: data.nome,
          family_name: data.cognome,
          ["custom:phone_number"]: data.telefono,
          ["custom:role"]: data.role,
          ["custom:codice_fiscale"]: data.codiceFiscale,
          ["custom:internal_id"]: data.internalId,
          ["custom:negozio_id"]: data.negozioId,
          ["custom:imprenditore_id"]: data.imprenditoreId,
          ["custom:ragione_sociale_id"]: data.ragioneSocialeId,
        },
      },
    });

    return res;
  }

```

### Log output

<details>

```
// Put your logs below this line


```

</details>

**Comment by nadetastic** [comment]
Hi @jimmysafe thank you for opening this issue. I am able to reproduce the same as well but it only seems to happen when the PreSign Up Lambda auto confirms the user - which I assume you are doing as well based on the the out put having ` signUpStep: 'DONE' ` - please let me know if I have this wrong.

Im discussing this with the team and will follow up soon.

**Comment by jimmysafe** [comment]
@nadetastic  yes that is correct, i am auto confirming the user with a PreSign Up Lambda:

```
module.exports.handler = async (event, context, callback) => {
  if (event.request.userAttributes['custom:role'] !== 'CLIENTE') {
    event.response.autoConfirmUser = true;
    event.response.autoVerifyEmail = true;
  }

  callback(null, event);
};

```

Also, checking your guys code i can see under `amplify-js/packages/auth/src/providers/cognito/apis/signUp.ts` that when the `signUpStep: 'DONE'` no `userId` is returned at all:

```
	} else if (isSignUpComplete(clientOutput) && !isAutoSignInStarted()) {
		return {
			isSignUpComplete: true,
			nextStep: {
				signUpStep: 'DONE',
			},
		};
```

it is weird tho, IMHO the user id should always be present, no matter of the signUpStep status.

Please let me know of any further development

Thanks

**Comment by nadetastic** [comment]
@jimmysafe thank you for clarifying! And for pointing out the source code where this seems to happen. Out of curiosity, could you expand on your use case for needing the userID on signUp when user is autoConfirmed?

**Comment by jimmysafe** [comment]
I have a built custom CMS platform where ADMIN users can manually register USERS.

When registering a user manually, the ADMIN goes through a multi step form, where he can:
- input new user data (register on Cognito)
- Input new user Store data (this will create a store in the DB)
- input new user Profile information (this will create a profile in the DB)
- ecc.. ecc..

At the moment when reaching step 2 (create user store) the DB expects a userId for the relation between Store and User, which i don't have because the signup function does not return userId 😂.

I'am using a workaround to fetch the user by email address before creating a Store, but that is  not a nice experience and i would like to remove it as it is very error prone.

Getting the userId back from the signup function is so much cleaner and it would make so much more sense.

**Comment by jimmysafe** [comment]
@nadetastic  any update on this?

**Comment by ashwinkumar6** [comment]
Hi @jimmysafe,  
Thank you for the detailed context on the usecase. We're currently looking into the issue. We appreciate your understanding as we work through this. If you have any more details or concerns to share, please feel free to do so.

**Comment by isaaccarrington** [comment]
For me this is the point at which I want to add the user to my application database

**Comment by isaaccarrington** [comment]
@ashwinkumar6  and @nadetastic is have the same problem with confirmSignUp. `userid` is not returned at all is `packages/auth/src/providers/cognito/apis/confirmSignUp.ts`. Do you want me to raise a seperate bug?

**Comment by israx** [comment]
hello @isaaccarrington . Returning `userId` in the `confirmSignUp` response would be more like a feature request. The reason is that the underlying service used by `confirmSignUp` doesn't return it. Amplify could store the userid from the signUp API in memory and then return it in `confirmSignUp` API, but take in mind that this behavior will not be consistent if the page is refreshed.

**Comment by ashwinkumar6** [comment]
Hi @jimmysafe, 
This is released in [6.0.27](https://github.com/aws-amplify/amplify-js/releases/tag/aws-amplify%406.0.27). Please let us know if you're run into any other issues!
